monai/apps/mmars/mmars.py

```diff
@@ -57,25 +101,55 @@ def download_mmar(item, mmar_dir=None, progress: bool = True):
 
     Args:
         item: the corresponding model item from `MODEL_DESC`.
+          Or when api is True, the substring to query NGC's model name field.
         mmar_dir: target directory to store the MMAR, default is mmars subfolder under `torch.hub get_dir()`.
         progress: whether to display a progress bar.
+        api: whether to query NGC and download via api
+        version: which version of MMAR to download.  default is -1 (latest).
 
     Examples::
         >>> from monai.apps import download_mmar
         >>> download_mmar("clara_pt_prostate_mri_segmentation_1", mmar_dir=".")
+        >>> download_mmar("prostate_mri_segmentation", mmar_dir=".", api=True)
+
 
     Returns:
         The local directory of the downloaded model.
+        If api is True, a list of local directories of downloaded models.
     """
-    if not isinstance(item, Mapping):
-        item = _get_model_spec(item)
     if not mmar_dir:
         get_dir, has_home = optional_import("torch.hub", name="get_dir")
         if has_home:
             mmar_dir = os.path.join(get_dir(), "mmars")
         else:
             raise ValueError("mmar_dir=None, but no suitable default directory computed. Upgrade Pytorch to 1.6+ ?")
 
+    if api:
+        model_dict = _get_all_ngc_models(item)
+        if len(model_dict) == 0:
+            raise ValueError(f"api query returns no item for pattern {item}.  Please change or shorten it.")
+        model_dir_list = list()
+        for k, v in model_dict.items():
+            url = f"https://api.ngc.nvidia.com/v2/models/{k}"
+            ver = v["latest"] if version == -1 else str(version)
+            download_url = f"{url}/versions/{ver}/zip"
+            model_dir = os.path.join(mmar_dir, v["name"])
+            download_and_extract(
+                url=download_url,
+                filepath=os.path.join(mmar_dir, f'{v["name"]}_{ver}.zip'),
+                output_dir=model_dir,
+                hash_val=None,
+                hash_type="md5",
+                file_type="zip",
+                has_base=False,
+                progress=progress,
+            )
+            model_dir_list.append(model_dir)
+        return model_dir_list
+
+    if not isinstance(item, Mapping):
+        item = _get_model_spec(item)
+
     model_dir = os.path.join(mmar_dir, item[Keys.ID])
     download_and_extract(
         url=item[Keys.URL],
```

Comment:
could you help update this line according to the line 135 notation?
```suggestion
        url=get_ngc_url(version, ...),
```
then `RemoteMMARKeys.URL`  such as this https://github.com/Project-MONAI/MONAI/blob/c20f70bf8c36c09630f7add95f823dbbcf1e9a73/monai/apps/mmars/model_desc.py#L45
could be ignored because it's hard-coded

Reply:
I added a simple helper function to form the url by combining Keys.NAME and user-provided version.

Reply:
thanks, could the version default to 1 for now? I can update `RemoteMMARKeys` to have a `VER` field later

Reply:
No problem.